### ISSUE TITLE
`AsBindGroup` derive example error

### ISSUE THREAD

**Issue by djeedai** [bevyengine/bevy#10510]
The [`AsBindGroup` derive example](https://docs.rs/bevy/latest/bevy/render/render_resource/trait.AsBindGroup.html#deriving) has 5 bindings but the corresponding WGSL code only has 4. I suspect this is not intended (or I didn't understand the example).

**Comment by superdump** [comment]
I wonder if it’s the combination of the values and the buffer fields that make the storage buffer binding…

**Comment by djeedai** [comment]
Yes possibly, but since I'm not sure and it's not clear I preferred opening a Bug anyway, and we should probably also clarify the doc in any case.

**Comment by djeedai** [comment]
The `buffer` argument of `#[storage]` is also not documented as far as I can see.
